Source/DynamicFeatureRegistry.swift

```diff
@@ -0,0 +1,46 @@
+// Copyright 2019, Oath Inc.
+// Licensed under the terms of the MIT license. See LICENSE file in https://github.com/yahoo/Override for terms.
+
+import Foundation
+
+/// DynamicFeatureRegistry a subclass of FeatureRegistry which provides an interface for using
+/// a collection of dynamic features instead of just pre-defined ones. When a feature is added,
+/// there is an attempt to apply an override state from the underlying data FeatureStore.
+@objc open class DynamicFeatureRegistry: FeatureRegistry {
+
+    /// A local store of the current dynamic features
+    private var dynamicFeatures: [String : AnyFeature] = [:]
+
+    override var features: [LabeledItem] {
+        let dynamicLabeledItems = dynamicFeatures.values.reduce(into: []) { (result, feature) in
+            result.append(LabeledFeatureItem(label: feature.key, feature: feature))
+        }
+        let staticFeatures = super.features
+        /// merge both dynamic and static features
+        let allFeatures = dynamicLabeledItems + staticFeatures
+        return allFeatures
+    }
+
+    /// This method is used to configure and store a dynamic feature. It will attempt to
+    /// assign an `override` value from the FeatureStore if it was previously set and then stores it
+    /// locally for future use.
+    /// - Parameter feature: A new feature
+    /// - Returns: The configured feature
+    public func configureDynamicFeature(feature: AnyFeature) -> AnyFeature {
+        // Bootstrap the last saved override value, if any
+        feature.override = featureStore[feature.key]
+
+        // Store feature in `dynamicFeatures` if needed.
+        if dynamicFeatures[feature.key] == nil {
+            dynamicFeatures[feature.key] = feature
+        }
+
+        if let baseFeature = feature as? BaseFeature {
+            baseFeature.overrideChangeHandler = featureDidChangeHandler
+            baseFeature.overrideChangeHandler = {[unowned self] feature in
+                self.featureStore[feature.key] = feature.override
+            }
```

Comment:
tl;dr: `weak self` is probably the way to go here.

Per [the comment](https://github.com/yahoo/Override/pull/27/files#diff-911730d490d6624dd03e9e6c31620d95b8f0b87632257cbca0e4fabd61dad159R75-R80) in FeatureRegistry below (L75-80), the `unowned` keyword was chosen because the corresponding object's lifecycle was provably tied to the lifecycle of `self`. I'm not so sure if that is true here because `feature` is currently owned by the caller. Although line 35 stores `feature` in `self.dynamicFeatures`, taking some ownership, the line 34 if-check might cause that to be bypassed resulting in a break in this chain.

In other words, if two callers pass two different `feature` parameters that have equivalent (as in ==) values in `feature.key`, one of those `feature` params will be mutated such that `overrideChangeHandler` contains a block with unowned self, but that instance will have no linkage to the lifecycle of self.

Reply:
Got it 👍